crates/spfs/src/graph/annotation.rs

```diff
@@ -95,10 +95,14 @@ impl<'a> AnnotationValue<'a> {
     pub fn legacy_encode(&self, mut writer: &mut impl std::io::Write) -> Result<()> {
         match self {
             AnnotationValue::String(v) => {
+                // Clippy doesn't realize `writer` can't be moved here.
+                #[allow(clippy::needless_borrows_for_generic_args)]
                 encoding::write_uint8(&mut writer, AnnotationValueKind::String as u8)?;
```

Comment:
I did a little digging into this one and it looks like the proper solution to avoid this lint is to re-borrow rather than creating a double-reference... did you try this pattern?

```suggestion
                encoding::write_uint8(&mut *writer, AnnotationValueKind::String as u8)?;
```

Reply:
I didn't try that. I'm traveling this weekend so if you want to take over to get this change in and merged feel free.

Reply:
Will do, I've moved the actual CI/builds as well now and will merge if things pass

Reply:
Thanks!